main.js

```diff
@@ -0,0 +1,329 @@
+
+  fiveDayForecastArr = getFiveDayForecastDayObjArr(data);
+  if(fiveDayForecastArr.length = 5){
+    for(let i = 0; i < fiveDayForecastArr.length; i++){
+      var forecastDayData = fiveDayForecastArr[i];
+      var forecast = {
+        city: forecastDayData.city,
+        dayOfWeek: forecastDayData.dayOfWeek,
+        temp: `${forecastDayData.temp}°`,
+        clouds: forecastDayData.clouds
+      }
+      fiveDayForecastsDOM.push(forecast);
+    }
+  } else if (fiveDayForecastArr.length = 6){
+    for(let i = 1; i < fiveDayForecastArr.length; i ++){
+      var forecastDayData = fiveDayForecastArr[i];
+      var forecast = {
+        city: forecastDayData.city,
+        dayOfWeek: forecastDayData.dayOfWeek, 
+        temp: `${forecastDayData.temp}°`,
+        clouds: forecastDayData.clouds
+      }
+      fiveDayForecastsDOM.push(forecast);
+    }
+  }
+  renderFiveDayForecast();
+}
+
+/**
+ * function creates HTML DOM template with values iterated from global variable
+ * array for the fiveDayForecastDOM element to be inserted into the HTML document 
+ */
+let renderFiveDayForecast = function() {
+  document.querySelector('.five-day-forecast').replaceChildren();
+
+  for(let i = 0; i < fiveDayForecastsDOM.length; i++){
+    let forecastForDay = fiveDayForecastsDOM[i];
+    let template = `
+      <div class="forecast-infobox column">
+        <h5 class="weather">${forecastForDay.clouds}</h5>
+        <h5 class="temperature">${forecastForDay.temp}</h5>
+        <div id="small-weather-icon">${selecSmallWeatherIcon(forecastForDay.clouds)}</div>
+        <h5 class="day-of-week">${forecastForDay.dayOfWeek}</h5>
+      </div>`;
+    
+    let htmlObject = document.createElement('div');
+    htmlObject.className = 'forecast';
+    htmlObject.innerHTML = template;
+  
+    document.querySelector('.five-day-forecast').insertAdjacentElement('beforeend', htmlObject);
+  }
+};
+
+/**
+ * 
+ * @param {JSON} data 
+ * @returns {Array} of day objects for five day forecast received from JSON API call
+ */
+let getFiveDayForecastDayObjArr = function(data){
+  let fiveDayForecastDays = [];
+  let allObjForecastEntriesArr = [];
+
+  data.list.forEach(function(item){
+    let dayOfWeek = getNameOfDay(item);
+    let dayObj = {
+      city: data.city.name,
+      dayOfWeek: dayOfWeek,
+      temp: Math.floor(item.main.temp),
+      clouds: item.weather[0].main
+    }
+    allObjForecastEntriesArr.push(dayObj);
+  });
+
+  fiveDayForecastDays = function(objArray){
+    return objArray.reduce(function(acc, dayOfWeek){
+      let exists = acc.find(function (item){
+        return dayOfWeek.dayOfWeek === item.dayOfWeek;
+      });
+      if(!exists){
+        acc.push(dayOfWeek);
+      }
+      return acc;
+    }, []);
+  }
+  return fiveDayForecastDays(allObjForecastEntriesArr);
+}
+
+/**
+ * 
+ * @param {JSON} item - item object from JSON object promise
+ * @returns {String} - string name of day of week from JSON object
+ */
+function getNameOfDay (item){

```

Comment:
Would recommend staying consistent with function expressions instead of using declarations